pytradfri/smart_task.py

```diff
@@ -183,51 +171,157 @@ def task_start_time_seconds(self):
-        return [TaskInfo(self._task, i) for i in range(len(self.raw))]
+        return [StartActionItem(
+            self._task,
+            i,
+            self.state,
+            self.api,
+            self.path) for i in range(len(self.raw))]
+
+    def set_dimmer_start_time(self, hour, minute):
+        """Set start time for task (hh:mm) in iso8601.
+
+        NB: dimmer starts 30 mins before time in app
+        """
+        #  This is to calculate the difference between local time
+        #  and the time in the gateway
+        d1 = dt.strptime(
+                self._gateway.get_gateway_info().current_time_iso8601,
+                '%Y-%m-%dT%H:%M:%S.%fZ')
+        d2 = dt.strptime(
+                dt.now().isoformat(),
+                '%Y-%m-%dT%H:%M:%S.%f')
+
+        diffMinutes = round(int(
+            time.mktime(d2.timetuple()) -
+            time.mktime(d1.timetuple())) / 60, 0)
+
+        newtime = dt(100, 1, 1, hour, minute, 00) + \
+            td(minutes=-diffMinutes)
+        command = {
+            ATTR_SMART_TASK_TRIGGER_TIME_INTERVAL:
+                [{
+                    ATTR_SMART_TASK_TRIGGER_TIME_START_HOUR: newtime.hour,
+                    ATTR_SMART_TASK_TRIGGER_TIME_START_MIN: newtime.minute
+                }]
+            }
+        self.set_values(command)
+
+    def set_values(self, command):
+        """Set values on task control."""
+        self._task.api('put', self._task.path, command)
+
+    @property
+    def raw(self):
+        """Return raw data that it represents."""
+        return self._task.raw[ATTR_START_ACTION]
+
+
+class StartAction:
+    """Class to control the start action-node."""
+
+    def __init__(self, start_action, api, path):
+        """Initialize StartAction class."""
+        self.start_action = start_action
+        self.api = api
+        self.path = path
+
+    @property
+    def state(self):
+        """Return state of start action task."""
+        return self.raw.get(ATTR_LIGHT_STATE)
+
+    @property
+    def devices(self):
+        """Return state of start action task."""
+        return [StartActionItem(
+            self.start_action,
+            i,
+            self.state,
+            self.api,
+            self.path,
+            self.raw) for i in range(
+                len(self.raw[ROOT_START_ACTION]))]
 
     @property
     def raw(self):
         """Return raw data that it represents."""
-        return self._task.raw[ATTR_START_ACTION][ROOT_START_ACTION]
+        return self.start_action.raw[ATTR_START_ACTION]
 
 
-class TaskInfo:
+class StartActionItem:
     """Class to show settings for a task."""
 
-    def __init__(self, task, index):
+    def __init__(self, task, index, state, api, path, raw):
         """Initialize TaskInfo."""
         self.task = task
         self.index = index
+        self.state = state
+        self.api = api
+        self.path = path
+        self._raw = raw
+
+    @property
+    def devices_json_list(self):
```

Comment:
We don't have JSON here, we just have Python objects. Let's remove `json` from the name.

Also, the name list is a bit weird because you're returning a dictionary.